google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/MavenBuildDeploymentSource.java

```diff
@@ -127,17 +115,17 @@ public File getFile() {
         new File(mavenProject.getBuildDirectory()).getPath() + File.separator
             + mavenProject.getFinalName();
 
-    XmlFile appEngineWebXml = AppEngineAssetProvider.getInstance()
-        .loadAppEngineStandardWebXml(project, Collections.singletonList(getModule()));
     AppEngineProjectService projectService = AppEngineProjectService.getInstance();
 
     // The environment will be null for newly deserialized deployment sources to ensure freshness.
     // In this case, we need to reload the environment.
     if (environment == null) {
-      environment = projectService.getModuleAppEngineEnvironment(appEngineWebXml);
+      environment = projectService.getModuleAppEngineEnvironment(getModule()).orElseThrow(
+          () -> new RuntimeException("No environment.")
```

Comment:
I believe that with these changes, the only way for a maven deployable to appear in the first place is with flex framework support in place. This means that the environment check should not return empty.

However, I think it will still be possible to break this and blow up the ide by first having a flex run config (and flex framework support), storing it with a maven deployable, and then _removing_ the flex facet manually. Upon deserialization I suspect this will blow up.

II